src/modules/m_connameban.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2016 Johanna Abrahamsson <johanna-a@users.noreply.github.com>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "inspircd.h"
+
+class ModuleConnameBan : public Module, public Whois::EventListener
+{
+	public:
+		Version GetVersion() CXX11_OVERRIDE
+		{
+			return Version("Conname 'n' - Connection name (class) ban",VF_OPTCOMMON);
+		}
+
+		ModuleConnameBan() : Whois::EventListener(this) {};
+
+		ModResult OnCheckBan(User *user, Channel *c, const std::string& mask) CXX11_OVERRIDE
+		{
+			LocalUser* localUser = IS_LOCAL(user);
+			if (localUser && (mask.length() > 2) && (mask[0] == 'n') && (mask[1] == ':'))
+			{
+				if (InspIRCd::Match(localUser->GetClass()->name, mask.substr(2)))
+					return MOD_RES_DENY;
+			}
+			return MOD_RES_PASSTHRU;
+		}
+
+		void OnWhois(Whois::Context& whois) CXX11_OVERRIDE
```

Comment:
I don't think coupling the extban with a whois line is a great idea, let's keep this simple and only have the extban and we'll figure out how's best to show the connect class in whois.